src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -627,6 +627,14 @@ Object byteBegin(RubyMatchData matchData, int index,
             }
         }
 
+        @TruffleBoundary
```

Comment:
I wonder if it's worth skipping `StringUtils.format` here and doing away with the boundary. This string can be trivially built with the `+` operator.

To be clear, I'm not saying you've done anything wrong here. I see you've matched what we're doing elsewhere in the file. And it's an exceptional case, so having the boundary isn't likely to cause problems. It's just that we can also remove it with minimal effort.

@eregon Thoughts on that?

Reply:
I think it's not worth spending time to optimize this likely-rare exceptional case.
And also not having the boundary would mean more code to PE.
Java String + is also not great to PE (I think can lead to some invoke dynamic stuff),
but the real killer here would be Integer#toString() which is definitely bad to PE (increase warmup, increase image size, unlikely to be faster PE'd than run behind a boundary).

However, this `@TruffleBoundary` on the specialization should be removed, because it hides the `throw` and prevents to PE the RaiseException constructor, getContext(), etc.
There is already a `@TruffleBoundary` on `StringUtils.format(` so that's enough.